kolibri/plugins/learn/assets/test/views/content-card-group-carousel.spec.js

```diff
@@ -0,0 +1,231 @@
+
+        wrapper.setProps({contents});
+
+
+        assert.ok(carousel.isFirstSet);
+      });
+      it('should be false when on the next set', function() {
+      });
+    });
+    describe('.isLastSet()', function() {
+      it('should be true when there are no more sets', function() {
+      });
+    });
+  });
+  describe('.template', function() {
+    function _singleButtonTest(){}
+    function _buttonDirectionTest(direction){}
+    function _buttonPlacementTest(side){}
+
+    describe('Multiple Sets', function() {
+      describe('First Set', function() {
+        beforeEach('Set up a multi-set carousel', function(){
+          const contents = makeContentData(15);
+          this.wrapper.setProps({contents});
+          this.wrapper.instance().elSize.width = this.wrapper.instance().contentCardWidth * 3;
+          // NOTE tried avoiding a nextTick in every test, didn't work. Dunno why.
+          // return Vue.nextTick();
+        });
+        it('contains the first content item passed in', function() {
+          const wrapper = this.wrapper;
+          const carousel = wrapper.instance();
+          const contents = makeContentData(15);
+          this.wrapper.setProps({contents});
+          this.wrapper.instance().elSize.width = this.wrapper.instance().contentCardWidth * 3;
+
+          return  carousel.$nextTick().then(()=>{
+            const firstCard = wrapper.first(card);
+            const {
+                title: cardTitle,
+                thumbnail: cardThumb,
+                kind: cardKind
+              } = firstCard.instance();
+
+            const {
+                title: contentTitle,
+                thumbnail: contentThumb,
+                kind: contentKind
+              } = carousel.contents[0];
+
+            assert.equals(cardTitle, contentTitle);
+            assert.equals(cardThumb, contentThumb);
+            assert.equals(cardKind, contentKind);
+          });
+        });
+
+        describe('LTR Languages', function() {
+          it('displays the first content item in the leftmost card', function() {
+            return Vue.nextTick().then(()=>{
+              const firstCard = this.wrapper.first(card);
+              assert.ok(firstCard.hasStyle('left', '0'), 'CSS property "left" is not set to 0');
+            });
+          });
+          describe('controls', function() {
+            it('contains a single button pointing right on the right');
+          });
+        });
+
+        describe('RTL Languages', function() {
+          it('displays the first content item in the rightmost card', function() {
+            const wrapper = this.wrapper;
+            const carousel = wrapper.instance();
+
+            // NOTE this lets the test pass... No idea why.
+            // return Vue.nextTick(()=>{
+            return Vue.nextTick().then(()=>{
+              const firstCard = wrapper.first(card);
+              assert.ok(firstCard.hasStyle('right', '0'), 'CSS property "right" is not set to 0');
```

Comment:
You are directly importing the component here, but all of the `left` to `right` css processing happens in our webpack pipeline. Best to not do tests of non-dynamic RTL css at this level.

Reply:
These are inline styles responding to isRtl - do we need the webpack pipeline for that too?

Reply:
No, any computed styles relying on the isRtl property (assuming that has been set to rtl within the test) should be testable, just those that are specified in the style block (which I think this is, from the component code).